.github/workflows/pr.yaml

```
@@ -76,7 +76,7 @@ jobs:
           pip install -r dev_tools/requirements/envs/pytest.env.txt
           pip install --no-deps -e .
       - run: |
-          python dev_tools/execute-notebooks.py
+          python dev_tools/execute-notebooks.py --n-workers=8
```

```suggestion
          python dev_tools/execute-notebooks.py --n-workers=$(nproc)
```

According to [GitHub's info about their current standard runners](https://docs.github.com/en/actions/reference/runners/github-hosted-runners#standard-github-hosted-runners-for-public-repositories), Linux runners get 4 virtual CPUs. If these notebook executions are primarily doing CPU-bound work and not I/O bound work, then it's _possible_ a value of > 4 here may actually be counter-productive. Also, the characteristics of the GitHub runners may change over time. So, instead of hard-coding this value to 8, what about using `nproc` here to get the number of processing units actually available to the CI job?

---

the thing is, these are pretty I/O bound. Usually half the time is spent spinning up the jupyter kernels (which run as little zmq servers) and other overhead

---

Got it. Yeah, for I/O bound I guess it's fine to exceed the number of CPUs.